tests/streaming/test_dataloader.py

```diff
@@ -330,3 +330,57 @@ def test_resume_dataloader_with_new_dataset(tmpdir):
     dataloader.load_state_dict(dataloader_state)
     for _ in dataloader:
         assert dataloader.current_epoch == 2, "Current epoch should be 2"
+
+
+@pytest.mark.timeout(120)
+def test_combined_dataset_dataloader_states(tmpdir):
```

Comment:
Thinking of moving this test to test_combined.py so that all the combined dataset tests stay together in one file.

Reply:
Good idea.